### PR TITLE
update GNU coreutils version in GnuTests workflow

### PR DESCRIPTION
**Pull request by jtracey**
Updates it to the next and latest release. This is necessary to pull the [fixed invalid-j test](https://github.com/coreutils/coreutils/commit/1d5dbb82030a2ca7aef606d36a9d04863240c7a8) that will, once all the outstanding join PRs get merged, allow join to pass all its GNU tests (#2634). Seems to make some other tests fail, so those might require more investigation, but presumably aren't because of bugs introduced in the GNU repo.

### CODE REVIEW

**Comment by sylvestre** [comment]
Ouch:

```
Warning: Congrats! The gnu test tests/ls/removed-directory is now passing!
Error: GNU test failed: tests/cp/attr-existing. tests/cp/attr-existing is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/ls/classify. tests/ls/classify is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/ls/selinux-segfault. tests/ls/selinux-segfault is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/ls/sort-width-option. tests/ls/sort-width-option is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/ls/zero-option. tests/ls/zero-option is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/misc/cksum. tests/misc/cksum is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/misc/cksum-a. tests/misc/cksum-a is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/misc/cksum-c. tests/misc/cksum-c is passing on 'main'. Maybe you have to rebase?
Error: GNU test failed: tests/misc/sm3sum. tests/misc/sm3sum is passing on 'main'. Maybe you have to rebase?
```

did you look at what is regressing?

**Comment by jtracey** [comment]
I had not, but did so now:
 - `tests/cp/attr-existing`:  now tests that cp does not remove files when `--attributes-only` is passed (specifically, files that would be overwritten were the flag not present)
  - `tests/ls/classify`: new test for new optional flag argument (`--classify` can now take `always`/`auto`/`never`)
  - `tests/ls/selinux-segfault`: adds check for "segfault when printing the security context of broken symlink targets"
  - `tests/ls/zero-option`: new test for new flag, `--zero`, that uses null for line endings instead of newline
  - `tests/misc/cksum`, `tests/misc/cksum-a`, `tests/misc/cksum-c`, `tests/misc/sm3sum`: a bunch of brand new tests, someone will have to dig into what all they're testing
  
So basically, no singular issue, just various updates to the tests. Some of these seem like new functionality that will have to be added, but at least the first one seems like an actual bug being caught.

**Comment by sylvestre** [comment]
Sounds good, could you please create issues to keep track of the regressions?
thanks

**Comment by jtracey** [comment]
I've updated my above comment with links to the respective issues.

**Comment by sylvestre** [comment]
amazing, thanks :)

**Comment by sylvestre** [comment]
let's regress then :)

**Comment by sylvestre** [comment]
I documented the regression here:
https://github.com/uutils/coreutils-tracking/commit/a10f9b0387c78f0c21b91e42085c3d8903ce3cd5
